app/main.py

```diff
@@ -1,34 +1,174 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self,
+                 row: int,
+                 column: int,
+                 is_alive: bool = True
+                 ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False
+                 ) -> None:
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = []
+        if start[1] == end[1]:
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+            for pos in range(start[0], end[0] + 1):
+                self.decks.append(Deck(pos, start[1]))
+        elif start[0] == end[0]:
+            for pos in range(start[1], end[1] + 1):
+                self.decks.append(Deck(start[0], pos))
+
+    def get_deck(self,
+                 row: int,
+                 column: int
+                 ) -> Deck:
+
+        return next(
+            deck for deck in self.decks if deck.row == row
+            and deck.column == column
+        )
+
+    def fire(self,
+             row: int,
+             column: int
+             ) -> None:
+
+        deck = self.get_deck(row, column)
+        deck.is_alive = False
+        if len([deck_ for deck_ in self.decks if not deck_.is_alive]) == len(
+                self.decks):
+            self.is_drowned = True
```

Comment:
You can use as follow:
```suggestion
            self.is_drowned = not any(deck.is_alive for deck in self.decks)
```